### PR TITLE
[SGP-5954] Support OAuth authentication for API requests

### PR DESCRIPTION
**Pull request by jrfernandes**
This PR updates the library to also support OAuth based authentication for API requests.

### CODE REVIEW

**Comment by abendig** [comment]
The library actually comes with test. Would you mind adding CI configuration (here or in a separate branch)?

File: meetup/api.py [hunk 1]
```diff
@@ -24,9 +24,13 @@ class MeetupClient(object):
     rate_limit_reset = 1
     last_response_time = None
 
-    def __init__(self, api_key):
+    request_headers = {}
```
Comment (abendig) [inline meetup/api.py]: I think it would be better to remove this default value and just set the value in the constructor.

File: meetup/api.py [hunk 2]
```diff
@@ -24,9 +24,13 @@ class MeetupClient(object):
...
+
+    def __init__(self, api_key=None, oauth_token=None):
         """ Find your api_key from https://secure.meetup.com/meetup_api/key/"""
         self.api_key = api_key
+        if oauth_token:
```
Comment (abendig) [inline meetup/api.py]: Would do something like this instead:
```
    self. request_headers = {'Authorization': 'Bearer %s' % oauth_token} if oauth_token else {}
```

**Comment by jrfernandes** [comment]
Cleaned up. As for adding CI here, in the interest of getting this out soon I would prefer to set that up separately. Thoughts?

**Comment by jrfernandes** [comment]
I changed the code so it only defines the headers if necessary, avoiding setting the default with an empty dictionary. I've also added tests. CI will come later if you're ok.

**Pull request merged**